pkg/rtc/participant.go

```diff
@@ -277,8 +277,17 @@ func NewParticipant(params ParticipantParams) (*ParticipantImpl, error) {
 	} else {
 		p.activeCounter.Add(2)
 	}
+
+	primaryPC.OnICEConnectionStateChange(func(state webrtc.ICEConnectionState) {
+		p.handleICEStateChange(true, state)
+	})
+	secondaryPC.OnICEConnectionStateChange(func(state webrtc.ICEConnectionState) {
+		p.handleICEStateChange(false, state)
+	})
```

Comment:
Listening for ICE connection state change as ICE could be connected, but peer connection is not connected (due to DTLS failure). The logic used below (will comment more there) is to check for short lived connections and restrict transport.